tiledb/cloud/taskgraphs/_codec.py

```diff
@@ -1,33 +1,54 @@
+import abc
 import base64
 import json
 from typing import Any, Dict, Optional, TypeVar, Union
 
+import attrs
 import cloudpickle
+import pyarrow
+import urllib3
 
 from tiledb.cloud._results import decoders
 from tiledb.cloud._results import visitor
+from tiledb.cloud.taskgraphs import types
 
 _T = TypeVar("_T")
 TOrDict = Union[_T, Dict[str, Any]]
-_SENTINEL_KEY = "__tdbudf__"
-_ESCAPE_CODE = "__escape__"
+
+_PICKLE_PROTOCOL = 4
+_ARROW_VERSION = pyarrow.MetadataVersion.V5
+SENTINEL_KEY = "__tdbudf__"
```

Comment:
this would be a good place for a link to a spec doc / spec.md / whatever we have as a "start here" place to read about the format